src/mscorlib/src/System/Threading/Tasks/Task.cs

```diff
@@ -2116,7 +2116,7 @@ internal ReadOnlyCollection<ExceptionDispatchInfo> GetExceptionDispatchInfos()
             Contract.Assert(exceptionsAvailable, "Must only be used when the task has faulted with exceptions.");
             return exceptionsAvailable ?
                 m_contingentProperties.m_exceptionsHolder.GetExceptionDispatchInfos() :
-                new ReadOnlyCollection<ExceptionDispatchInfo>(new ExceptionDispatchInfo[0]);
+                new ReadOnlyCollection<ExceptionDispatchInfo>(Array.Empty<ExceptionDispatchInfo>());
```

Comment:
This path should never bit hit (see the earlier assert).  No need to optimize code that'll never run.  Let's revert this.